pulsar-common/src/main/proto/PulsarApi.proto

```diff
@@ -22,6 +22,20 @@ package pulsar.proto;
 option java_package = "org.apache.pulsar.common.api.proto";
 option optimize_for = LITE_RUNTIME;
 
+message Schema {
+	required string name = 1;
+	required bytes version = 2;
+	required bytes schema_data = 3;
```

Comment:
Does the schema data itself have a schema? If we're supporting json, arvo, pb, etc, shouldn't we have a separate field to give the reader a hint of how they need to parse the schema?

Reply:
I tried to address this in a question posed on the original reference PR. I had originally specced this out with that intent but after much thought I decided not to add it. My intent was that the schema registry should just store a "Schema" no matter what it is, it does not need to know what it contains. The schema only really means something to the end systems (producer/consumer) and by extension the broker. We can allow the broker to "plug-in" modules to allow semantic parsing of the schema as needed. This way we can support any number of crazy business cases.

Reply:
I agree that storing the schema data is agnostic to the format, however these protobufs are relative to the interactions between client and broker. 

In this case, the producer might use something like `Schema.avroOf(MyClass.class)` or `Schema.jsonOf(MyClass.class)`. In this case,  don't we need to communicate to the server what kind of schema the client is using?

Reply:
The properties list is available to store that info. This makes the most sense to me but I'm willing to change it if you are dead set on specifying this information statically.